### PR TITLE
chore(evg): drop zSeries for all but rhel, reintroduce rhel72

### PR DESCRIPTION
**Pull request by mbroadst**
NODE-1679

### CODE REVIEW

File: .evergreen/config.yml [hunk 1]
```diff
@@ -456,14 +446,9 @@ axes:
...
       # NOTE: reenable when these are actually running 7.2, or we release a 7.4 rpm
-      # - id: rhel72-zseries-test
-      #   display_name: "RHEL 7.2 (zSeries)"
-      #   run_on: rhel72-zseries-test
-      #   batchtime: 10080  # 7 days
-
-      - id: suse12-zseries-test
-        display_name: "SUSE 12 (zSeries)"
-        run_on: suse12-zseries-test
+      - id: rhel72-zseries-test
```
Comment (kvwalker) [inline .evergreen/config.yml]: Should this be `rhel67-zseries-test`, which is what the [drivers ticket ](https://jira.mongodb.org/browse/DRIVERS-469) says?

Comment (mbroadst) [reply .evergreen/config.yml]: yep! nice catch thanks

Comment (mbroadst) [reply .evergreen/config.yml]: hm looks like we can't actually build over there due to gcc version, let's let this PR sit until someone on dev prod can update on that

Comment (mbroadst) [reply .evergreen/config.yml]: @kvwalker it was determined we should stick with rhel72, reverted

**Pull request merged**